### PR TITLE
[FLINK-35092][cdc][starrocks] Add starrocks integration test cases

### PR DESCRIPTION
**Pull request by yuxiqian**
This closes [FLINK-35092](https://issues.apache.org/jira/browse/FLINK-35092).

Currently, Flink CDC does not have any integrated tests for StarRocks pipeline connector which runs on real docker container. Adding one should help improving pipeline connectors' reliability.

### CODE REVIEW

**Comment by yuxiqian** [comment]
cc @banmoy

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/resources/log4j2-test.properties [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 # Set root logger level to OFF to not flood build logs
 # set manually to INFO for debugging purposes
-rootLogger.level=OFF
+rootLogger.level=INFO
```
Comment (banmoy) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/resources/log4j2-test.properties:18]: set it to `OFF`

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/utils/StarRocksContainer.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/** Docker container for StarRocks. */
+public class StarRocksContainer extends JdbcDatabaseContainer<StarRocksContainer> {
+
+    private static final String DOCKER_IMAGE_NAME = "starrocks/allin1-ubuntu:3.1.10";
```
Comment (banmoy) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/utils/StarRocksContainer.java:35]: can use the newest 3.2.6

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/utils/StarRocksContainer.java [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    // FE node exposed ports
+    public static final int FE_HTTP_PORT = 8030;
+    public static final int FE_HTTP_SERVICE_PORT = 8080;
```
Comment (banmoy) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/utils/StarRocksContainer.java:39]: what's 8080 used for?

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/utils/StarRocksSinkTestBase.java [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+            new MiniClusterWithClientResource(
+                    new MiniClusterResourceConfiguration.Builder()
+                            .setNumberTaskManagers(1)
+                            .setNumberSlotsPerTaskManager(DEFAULT_PARALLELISM)
+                            .setRpcServiceSharing(RpcServiceSharing.DEDICATED)
+                            .withHaLeadershipControl()
+                            .build());
+
+    @BeforeClass
+    public static void startContainers() {
```
Comment (banmoy) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/utils/StarRocksSinkTestBase.java:80]: checking `SHOW BACKENDS;` is more stable to make sure the cluster is ready?

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/utils/StarRocksSinkTestBase.java [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+            if (rs.getExitCode() != 0) {
+                throw new RuntimeException("Failed to create database." + rs.getStderr());
+            }
+        } catch (Exception e) {
+            throw new RuntimeException("Failed to create database.", e);
+        }
+    }
+
+    public static void createTable(
```
Comment (banmoy) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/utils/StarRocksSinkTestBase.java:150]: construct the ddl outside, and just deliver the ddl string to make the method general to be reused?

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksPipelineITCase.java [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+                                new Object[] {17, 3.14, BinaryStringData.fromString("Doris Day")}),
+                        generator.generate(
+                                new Object[] {
+                                    17, 6.28, BinaryStringData.fromString("Doris Day")
+                                })));
+    }
+
+    @Test
+    public void testValuesToStarRocks() throws Exception {
+        //        Thread.sleep(100000000);
```
Comment (banmoy) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksPipelineITCase.java:141]: useless ?

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksPipelineITCase.java [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+        BinaryRecordDataGenerator generator =
+                new BinaryRecordDataGenerator(
+                        RowType.of(DataTypes.INT(), DataTypes.DOUBLE(), DataTypes.VARCHAR(17)));
+
+        return Arrays.asList(
+                new CreateTableEvent(tableId, schema),
+                DataChangeEvent.insertEvent(
+                        tableId,
+                        generator.generate(
+                                new Object[] {17, 3.14, BinaryStringData.fromString("Doris Day")})),
```
Comment (banmoy) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksPipelineITCase.java:110]:
```suggestion
                                new Object[] {17, 3.14, BinaryStringData.fromString("StarRocks Day")})),
```

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksPipelineITCase.java [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+
+        Sink<Event> starRocksSink =
+                ((FlinkSinkProvider) createStarRocksDataSink(config).getEventSinkProvider())
+                        .getSink();
+        stream.sinkTo(starRocksSink);
+
+        env.execute("Values to StarRocks Sink");
+
+        List<String> actual = fetchTableContent(tableId, 3);
+        List<String> expected = Arrays.asList("17 | 6.28 | Doris Day", "21 | 1.732 | Disenchanted");
```
Comment (banmoy) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksPipelineITCase.java:164]:
```suggestion
        List<String> expected = Arrays.asList("17 | 6.28 | StarRocks Day", "21 | 1.732 | Disenchanted");
```

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksPipelineITCase.java [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+                                new Object[] {
+                                    19, 2.718, BinaryStringData.fromString("Que Sera Sera")
+                                })),
+                DataChangeEvent.updateEvent(
+                        tableId,
+                        generator.generate(
+                                new Object[] {17, 3.14, BinaryStringData.fromString("Doris Day")}),
+                        generator.generate(
+                                new Object[] {
+                                    17, 6.28, BinaryStringData.fromString("Doris Day")
```
Comment (banmoy) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksPipelineITCase.java:135]:
```suggestion
                                    17, 6.28, BinaryStringData.fromString("StarRocks Day")
```

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksPipelineITCase.java [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+                DataChangeEvent.deleteEvent(
+                        tableId,
+                        generator.generate(
+                                new Object[] {
+                                    19, 2.718, BinaryStringData.fromString("Que Sera Sera")
+                                })),
+                DataChangeEvent.updateEvent(
+                        tableId,
+                        generator.generate(
+                                new Object[] {17, 3.14, BinaryStringData.fromString("Doris Day")}),
```
Comment (banmoy) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/StarRocksPipelineITCase.java:132]:
```suggestion
                                new Object[] {17, 3.14, BinaryStringData.fromString("StarRocks Day")}),
```

**Comment by banmoy** [comment]
@yuxiqian Thanks for the work. Left some comments.  Also cc @lvyanquan

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/utils/StarRocksContainer.java (hunk 3 above)
Comment (yuxiqian) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-starrocks/src/test/java/org/apache/flink/cdc/connectors/starrocks/sink/utils/StarRocksContainer.java:38]: StarRocks all-in-one docker image exposes 8080 as service port [in this PR](https://github.com/StarRocks/starrocks/pull/28240/). Removed redundant port mappings.

**Comment by yuxiqian** [comment]
Thanks for @banmoy's kindly review, addressed comments above.

**Review by banmoy — APPROVED** [review]

**Review by lvyanquan — APPROVED** [review]
LGTM.

**Comment by yuxiqian** [comment]
Fixed in #3348.

**Pull request closed without merging**